sd/eureka/registrar.go

```diff
@@ -0,0 +1,74 @@
+package eureka
+
+import (
+	"fmt"
+	"time"
+
+	fargo "github.com/hudl/fargo"
+
+	"github.com/go-kit/kit/log"
+)
+
+// Registrar maintains service instance liveness information in Eureka.
+type Registrar struct {
+	client   Client
+	instance *fargo.Instance
+	logger   log.Logger
+	quit     chan bool
+}
+
+// NewRegistrar returns an Eureka Registrar acting on behalf of the provided
+// Fargo instance.
+func NewRegistrar(client Client, i *fargo.Instance, logger log.Logger) *Registrar {
+	return &Registrar{
+		client:   client,
+		instance: i,
+		logger:   log.With(logger, "service", i.App, "address", fmt.Sprintf("%s:%d", i.IPAddr, i.Port)),
+	}
+}
+
+// Register implements sd.Registrar interface.
+func (r *Registrar) Register() {
+	if err := r.client.Register(r.instance); err != nil {
+		r.logger.Log("err", err)
+	} else {
+		r.logger.Log("action", "register")
+	}
+
+	if r.instance.LeaseInfo.RenewalIntervalInSecs > 0 {
+		// User has opted for heartbeat functionality in Eureka.
+		if r.quit == nil {
+			r.quit = make(chan bool)
+			go r.loop()
+		}
```

Comment:
The behaviour is to make periodic heartbeat calls to the Eureka registry after initial registration. By default Eureka will expect a heartbeat every 30s, and 3 missed heartbeats will result in the instance being expired. This is why I felt it was a must-have, and while Fargo makes available a Heartbeat RPC for the correct Eureka endpoint, it does not provide the periodic functionality. 

Also, these heartbeat periods and expiry thresholds are decided when setting up a Eureka server cluster and will vary in the wild. To cater for this I made the heartbeat interval configurable in the go-kit implementation using the instance's `LeaseInfo.RenewalIntervalInSecs` (part of the Eureka client spec).

Regarding how the behaviours compare, I had a look around and found similar functionality (I think) in [`sd/etcd/registrar.go`](https://github.com/go-kit/kit/blob/master/sd/etcd/registrar.go) with its `TTLOption` and mutex gated heartbeat loop. However, all that said - I had a play around there and I can achieve the same with a WaitGroup rather than Mutex. Please check the latest when you get a chance.